src/pybind/mgr/dashboard/tools.py

```diff
@@ -828,3 +828,13 @@ def find_object_in_list(key, value, iterable):
         if key in obj and obj[key] == value:
             return obj
     return None
+
+
+def merge_list_of_dicts_by_key(target_list: list, source_list: list, key: str):
+    target_list = {d[key]: d for d in target_list}
+    for sdict in source_list:
+        if bool(sdict):
+            if sdict[key] in target_list:
+                target_list[sdict[key]].update(sdict)
+    target_list = [value for value in target_list.values()]
+    return target_list
```

Comment:
Note we have a similar function already: 

https://github.com/ceph/ceph/blob/f938712b7c81f9e004827d1dc2fe0f005ee1b3b0/src/pybind/mgr/mgr_util.py#L402

Reply:
But this merging is based on a key and it merges list[dict]. Maybe should I add this to mgr_util?

Reply:
you've re-implemented merge_dicts in your merge_list_of_dicts_by_key funciton

Reply:
Not exactly, I can't directly consume tht func , I still need some modification after tht ofr e.g. to return all as a list and tht wld require another method, so I implemented all in 1 method, but I can try using `merge_dicts` in my current method.